component/pyroscope/ebpf/ebpfspy/symtab/gcache.go

```diff
@@ -0,0 +1,166 @@
+package symtab
+
+import (
+	"fmt"
+
+	lru "github.com/hashicorp/golang-lru/v2"
+)
+
+type Resource interface {
+	comparable
+	Refresh()
+	Cleanup()
+}
+
+type GCache[K comparable, V Resource] struct {
+	options GCacheOptions
+
+	roundCache map[K]*entry[V]
+	lruCache   *lru.Cache[K, *entry[V]]
+
+	round int
+}
+type entry[V Resource] struct {
+	v     V
+	round int
+}
+
+type GCacheOptions struct {
+	Size       int
+	KeepRounds int
```

Comment:
What does `KeepRounds` do ? May be add a comment also about size that is it in amount of element not size of them.